GitVersionCore.Tests/Helpers/GitTestExtensions.cs

```diff
@@ -39,6 +37,48 @@ public static Commit[] MakeCommits(this IRepository repository, int numCommitsTo
             .ToArray();
     }
 
+    public static Commit CreateFileAndCommit(this IRepository repository, string relativeFileName, DateTimeOffset dateTimeOffset = default(DateTimeOffset))
+    {
+        if (dateTimeOffset == default(DateTimeOffset))
+        {
+            dateTimeOffset = DateTimeOffset.Now;
+        }
+
+        var randomFile = Path.Combine(repository.Info.WorkingDirectory, relativeFileName);
+        if (File.Exists(randomFile))
+        {
+            File.Delete(randomFile);
+        }
+
+        File.WriteAllText(randomFile, Guid.NewGuid().ToString());
+
+        // GHK: 2015-01-18: I know it's very ugly, but somehow we need to retry here otherwise "there is nothing to commit"
+        var retryCount = 3;
```

Comment:
I have changed config based versioning branch to increment the commit time for each commit - https://github.com/ParticularLabs/GitVersion/commit/14487a0dd6b5b13b05dbff798e2132a54b06c49a#diff-c7ff189f687fa58758611ef26120e6dcR8

Not sure if this would fix the issue.